vmm/src/cpu.rs

```diff
@@ -518,12 +521,6 @@ impl BusDevice for CpuManager {
                     {
                         state.inserting = false;
                     }
-                    // Ditto for removal
-                    if (data[0] & (1 << CPU_REMOVING_FLAG) == 1 << CPU_REMOVING_FLAG)
```

Comment:
Hhm, I think we might still want to keep the ACPI pending removal flag - my only worry is if the kernel gets confused that is has acknowledged the removal and then the removal bit doesn't get cleared when it checks again. Maybe we need to track another bit of state :-(

Reply:
Okay. Let me think a better way out.

Reply:
I think it's fine to keep the existing `removing` bool. And add a `Arc<AtomicBool>>` called `pending_removal`. Set both on the request to remove, let the guest clear the `removing` as it sees fit but use the atomic one to check if a subsequent removal should be rejected.

Reply:
Thx for ur suggestions, that sounds make sense and I will update the patch.